Remove Pandas from experiment summary

MattToast opened the pull request:
- Refactor `Experiment.summary` method by removing usage of a pandas dataframe and instead utilizing tabulate to format the summary table
- Remove pandas as requirement for SmartSim
- Add tabulate as as requirement for SmartSim

Spartee reviewed (changes requested):
hey Matt! great stuff on the PR! 

A couple comments

theres a test in ``on_wlm/test_simple_entity_launch.py`` that tests the summary function and will need to be changed. In addition, we should add another test in the `test_experiment.py`` file to test another simple run.

MattToast commented:
Changes:
- Refactored `on_wlm/test_simple_entity_launch.py` to work with the summary `str` return type
- Added a test to `test_experiment.py` to test summary method without WLM integrations

Spartee commented on smartsim/experiment.py:
```diff
@@ -467,35 +467,35 @@ def summary(self):
 
-        :return: pandas Dataframe of ``Experiment`` history
-        :rtype: pd.DataFrame
+        :return: tabulate string of ``Experiment`` history
```
Lets allow ``format`` as a parameter here and pass to the tabulate function. Also should add in https://github.com/astanin/python-tabulate#table-format to the docstring so that people know what options are available.

Spartee commented on smartsim/experiment.py:
```diff
@@ -467,35 +467,35 @@ def summary(self):
+                        job.history.returns[run],
+                    ]
+                )
+        return tabulate(values, headers, showindex=True, tablefmt="plain")
```
lets set the default to ``github``. so the added parameter will be a keyword arg like ``format=github`` and then that input parameter will be passed here.

Spartee reviewed (changes requested):
Small changes! Otherwise looking great!

MattToast commented:
Changes:
- Added `format` as a parameter to `Experiment.summary` and passed value to call to `tabulate`
- Updated docstring with link to formats

Spartee reviewed (approved):
LGTM, lets chat before you merge!
